src/core/utils/addons.js

```diff
@@ -62,3 +63,51 @@ export const getFileHash = ({
 
   return undefined;
 };
+
+export function removeUndefinedProps(object: Object): Object {
+  const newObject = {};
+  Object.keys(object).forEach((key) => {
+    if (typeof object[key] !== 'undefined') {
+      newObject[key] = object[key];
+    }
+  });
+  return newObject;
+}
+
+export const getAddonJsonLinkedData = ({
+  addon,
+}: {|
+  addon: AddonType,
+|}): Object | null => {
```

Comment:
can we be any more specific than Object or could it be too variable to have something more specific than this?

Reply:
I don't think we can since it is not a fixed data structure.